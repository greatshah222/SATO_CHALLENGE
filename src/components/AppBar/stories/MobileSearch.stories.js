import React from 'react';
import { storiesOf } from '@storybook/react';
import { muiTheme } from 'storybook-addon-material-ui';
import { withKnobs, select, boolean } from '@storybook/addon-knobs';
import { withBackgrounds } from '@storybook/addon-backgrounds';

import MobileSearch from '../MobileSearch.js';
import theme from '../../../theme';
import { action } from '@storybook/addon-actions';

export const bg = withBackgrounds([{ name: 'light grey', value: '#f2f2f2', default: true }]);

storiesOf('MobileSearch', module)
  .addDecorator(bg)
  .addDecorator(muiTheme(Object.values(theme)))
  .addDecorator(withKnobs)
  .addWithJSX('default', () => (
    <MobileSearch
      searchTerm={''}
      searchPlaceholder={'Etsi...'}
      closeSearch={action('Close search')}
      handleSearch={action('Handle search')}
    />
  ));
