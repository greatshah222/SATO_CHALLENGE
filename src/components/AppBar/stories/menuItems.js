import { action } from '@storybook/addon-actions';

export const accountItems = [
  {
    label: 'Omasato',
    handler: action('Omasato clicked'),
    variant: 'button',
  },
  {
    label: 'Minun kotini',
    handler: action('Minun kotini clicked'),
    variant: 'buttonMenuItem',
    icon: 'home',
  },
  {
    label: 'Viestit',
    handler: action('Viestit clicked'),
    variant: 'buttonMenuItem',
    icon: 'message',
  },
  {
    label: 'Asiakasedut',
    handler: action('Asiakasedut clicked'),
    variant: 'buttonMenuItem',
    active: true,
    icon: 'benefit',
  },
  {
    label: 'Omat tiedot',
    handler: action('Omat tiedot clicked'),
    variant: 'buttonMenuItem',
    icon: 'loggedInUser',
    iconContent: 'AK',
  },
  {
    label: 'Kirjaudu ulos',
    handler: action('Kirjaudu ulos clicked'),
    variant: 'buttonMenuItem',
    icon: 'logout',
  },
];

export const faqItems = [
  {
    id: 9,
    label: 'Kodin etsiminen',
    handler: action('Kodin etsiminen clicked'),
    variant: 'buttonMenuItem',
    items: [],
  },
];

export const infoItems = [
  {
    id: 6,
    label: 'Usein kysyttyä',
    handler: action('Usein kysyttyä clicked'),
    variant: 'buttonMenuItem',
    items: faqItems,
  },
  {
    id: 7,
    label: 'Ajankohtaista',
    handler: action('Ajankohtaista clicked'),
    variant: 'buttonMenuItem',
    active: true,
  },
  {
    id: 8,
    label: 'Kotona',
    handler: action('Kotona'),
    variant: 'buttonMenuItem',
  },
];

export const links = [
  {
    label: 'Etusivu',
    active: true,
    route: '/',
  },
  {
    label: 'Sijoittajille',
    handler: action('Sijoittajille'),
    type: 'smallOnMobile',
  },
  {
    id: 10,
    small: true,
    executeHandler: true,
    label: 'Medialle',
    handler: action('Medialle'),
    type: 'smallOnMobile',
    items: [
      {
        label: 'Mediapankki',
        variant: 'buttonMenuItem',
        handler: action('mediapankki'),
      },
      {
        label: 'Etsitkö asiantuntijaa?',
        variant: 'buttonMenuItem',
        handler: action('asintuntija'),
      },
    ],
  },
  {
    label: 'SATO yrityksenä',
    handler: action('SATO yrityksenä'),
    type: 'smallOnMobile',
  },
];

export const localeLink = {
  label: 'In English',
  handler: action('Change locale'),
};

export const buttons = [
  {
    id: 1,
    label: 'Löydä koti',
    handler: action('Löydä koti'),
    variant: 'button',
    active: true,
  },
  {
    id: 2,
    label: 'Tietoa asumisesta',
    handler: action('Tietoa asumisesta'),
    variant: 'button',
    items: infoItems,
    active: true,
  },
  {
    id: 4,
    label: 'Ota yhteyttä',
    handler: action('Ota yhteyttä'),
    variant: 'button',
  },
];
