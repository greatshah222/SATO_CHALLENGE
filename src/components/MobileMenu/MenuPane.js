import PropTypes from 'prop-types';
import React from 'react';
import Text from '../Text';
import Slide from '@material-ui/core/Slide';
import ChevronRight from '../../icons/ChevronRight';
import ArrowLeft from '../../icons/ArrowLeft';
import { withStyles } from '@material-ui/core/styles';
import * as tokens from '../../.generated/styles';
import classNames from 'classnames';
import NavigationList, { NavigationListItem } from '../NavigationList';
import Home from '../../icons/Home';
import HomeSearch from '../../icons/HomeSearch';
import Message from '../../icons/Message';
import Benefit from '../../icons/Benefit';
import LoggedInUser from '../../icons/LoggedInUser';
import Logout from '../../icons/Logout';

import CircularProgress from '@material-ui/core/CircularProgress';

const styles = theme => ({
  list: {
    position: 'absolute',
    left: 0,
    right: 0,
  },
  button: {
    fontSize: tokens.fontSize5,
    width: '100%',
  },
  buttonSmallFont: {
    fontSize: tokens.fontSize2,
  },
  space: {
    width: tokens.spacingMedium,
    display: 'inline-block',
  },
  listItem: {
    paddingLeft: tokens.spacing4Xl,
    paddingRight: 0,
  },
  icon: {
    paddingRight: '1rem',
    color: theme.palette.primary.main,
  },
});

const icons = {
  home: Home,
  message: Message,
  benefit: Benefit,
  loggedInUser: LoggedInUser,
  logout: Logout,
  homeSearch: HomeSearch,
};

class MenuPane extends React.Component {
  handleClick = link => () => {
    if (link.executeHandler) {
      link.handler();
    }
    this.props.goToChild(link.id);
  };

  handleLinkClick = cb => () => {
    this.props.toggleMenu();
    cb();
  };

  getIcon = (icon, content) => {
    const Icon = icons[icon];
    return Icon ? <Icon className={this.props.classes.icon}>{content}</Icon> : null;
  };

  getItem = link => {
    if (link.items) {
      return (
        <NavigationListItem onClick={this.handleClick(link)} type={link.type} key={link.id} active={link.active}>
          <Text
            variant={link.variant}
            className={classNames(this.props.classes.button, {
              [this.props.classes.buttonSmallFont]: link.small,
            })}
            color={'inherit'}
          >
            {`${link.small ? '\u00A0' : ''}${link.label}`}
            <ChevronRight style={{ float: 'right' }} />
          </Text>
        </NavigationListItem>
      );
    }
    return (
      <NavigationListItem
        onClick={this.handleLinkClick(link.handler)}
        type={link.type}
        key={link.id}
        active={link.active}
      >
        {this.getIcon(link.icon, link.iconContent)}&nbsp;
        <Text variant={link.variant || 'body1'} className={classNames({ [this.props.classes.button]: !!link.variant })}>
          {link.label}
        </Text>
      </NavigationListItem>
    );
  };

  getItems = items => items.map(link => this.getItem(link));

  render() {
    const {
      visible,
      direction,
      goBack,
      hasParent,
      classes,
      currentItem: { label, handler, id, items, lazy },
      backLabel,
      animate,
      className,
      ...rest
    } = this.props;

    const navItems = this.getItems(items || []);
    const isEmpty = navItems.length === 0;
    if (label) {
      navItems.unshift(this.getItem({ label, handler, id, variant: 'button' }));
    }

    const List = (
      <NavigationList className={classNames(classes.list, className)} {...rest}>
        {hasParent && (
          <NavigationListItem onClick={goBack} key={'back'} leftAligned noLeftMargin>
            <Text variant={'body1'}>
              <ArrowLeft style={{ float: 'left' }} />
              <span className={classes.space} />
              {backLabel}
            </Text>
          </NavigationListItem>
        )}
        {navItems}
        {lazy && isEmpty && (
          <NavigationListItem>
            <CircularProgress color="primary" />
          </NavigationListItem>
        )}
      </NavigationList>
    );

    if (!animate) {
      return List;
    }
    return (
      <Slide direction={direction} in={visible} unmountOnExit={true}>
        {List}
      </Slide>
    );
  }
}

MenuPane.propTypes = {
  direction: PropTypes.oneOf(['left', 'right']).isRequired,
  visible: PropTypes.bool.isRequired,
  hasParent: PropTypes.bool.isRequired,
  goToChild: PropTypes.func.isRequired,
  goBack: PropTypes.func.isRequired,
  classes: PropTypes.object,
  toggleMenu: PropTypes.func.isRequired,
  currentItem: PropTypes.object.isRequired,
  backLabel: PropTypes.string.isRequired,
  animate: PropTypes.bool.isRequired,
};

export default withStyles(styles)(MenuPane);
