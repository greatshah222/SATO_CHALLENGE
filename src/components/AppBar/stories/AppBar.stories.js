import React from 'react';
import { storiesOf } from '@storybook/react';
import { muiTheme } from 'storybook-addon-material-ui';
import { withKnobs, select, boolean } from '@storybook/addon-knobs';
import { withBackgrounds } from '@storybook/addon-backgrounds';
import { accountItems, buttons, links, localeLink } from './menuItems';

import AppBar from '../index.js';
import theme from '../../../theme';
import { action } from '@storybook/addon-actions';

export const bg = withBackgrounds([{ name: 'light grey', value: '#f2f2f2', default: true }]);

storiesOf('AppBar', module)
  .addDecorator(bg)
  .addDecorator(muiTheme(Object.values(theme)))
  .addDecorator(withKnobs)
  .addWithJSX('unsigned', () => (
    <AppBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={false}
      links={links}
      localeLink={localeLink}
      defaultBackLabel={'Takaisin'}
      accountBtnLabel={'Kirjaudu'}
      accountBtnHandler={action('Login clicked')}
      handleSearch={() => {}}
      searchSuggestions={[]}
    />
  ))
  .addWithJSX('logged in', () => (
    <AppBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={true}
      links={links}
      localeLink={localeLink}
      defaultBackLabel={'Takaisin'}
      accountBtnLabel={'Omasato'}
      accountBtnHandler={action('Omasato clicked')}
      handleSearch={() => {}}
      searchSuggestions={[]}
    />
  ))
  .addWithJSX('playground', () => (
    <AppBar
      accountItems={accountItems}
      buttons={buttons}
      links={links}
      localeLink={localeLink}
      loggedIn={boolean('Logged in', false, 'General')}
      defaultBackLabel={'Takaisin'}
      accountBtnLabel={'Kirjaudu'}
      accountBtnHandler={action('Login clicked')}
      handleSearch={() => {}}
      searchSuggestions={[]}
    />
  ));
