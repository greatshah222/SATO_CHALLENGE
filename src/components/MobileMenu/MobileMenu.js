import PropTypes from 'prop-types';
import React from 'react';
import MenuPane from './MenuPane';
import { findById, findParentById } from '../../utils';
import isEqual from 'lodash/isEqual';

class MobileMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentItem: props.content,
      previousItem: null,
      toChild: true,
      backLabel: props.defaultBackLabel,
    };
  }

  // this function is here to check if current items subitems was updated
  // setState must be called manually to rerender lazy loaded subitems
  componentDidUpdate(prevProps, prevState) {
    if (!isEqual(prevProps.content, this.props.content)) {
      if (!isEqual(prevState.currentItem.items, this.props.content.items)) {
        if (
          prevState.previousItem !== null &&
          this.state.previousItem !== null &&
          isEqual(prevState.previousItem, this.state.previousItem) &&
          isEqual(prevState.currentItem, this.state.currentItem)
        ) {
          const updatedItems = findById(prevState.currentItem.id, this.props.content.items);
          if (updatedItems) {
            this.setState({ currentItem: updatedItems });
          }
        }
      }
    }
  }

  goToChild = id =>
    this.setState({
      previousItem: this.state.currentItem,
      currentItem: findById(id, this.state.currentItem.items),
      toChild: true,
      backLabel: this.state.currentItem.label || this.props.defaultBackLabel,
      previousBackLabel: this.state.backLabel,
    });

  goBack = () => {
    const parent = findParentById(this.state.currentItem.id, this.props.content);
    const grandParent = findParentById(parent.id, this.props.content);
    this.setState({
      currentItem: parent,
      previousItem: this.state.currentItem,
      toChild: false,
      backLabel: grandParent && grandParent.label ? grandParent.label : this.props.defaultBackLabel,
      previousBackLabel: this.state.backLabel,
    });
  };

  render() {
    const { open, toggleMenu, ...rest } = this.props;
    const { currentItem, previousItem, toChild, backLabel, previousBackLabel } = this.state;

    return (
      <div>
        {previousItem && (
          <MenuPane
            direction={toChild ? 'right' : 'left'}
            visible={false}
            key={previousItem.id}
            goToChild={this.goToChild}
            goBack={this.goBack}
            hasParent={previousItem.id !== 'root'}
            toggleMenu={toggleMenu}
            currentItem={previousItem}
            backLabel={previousBackLabel}
            animate={true}
            {...rest}
          />
        )}
        <MenuPane
          direction={toChild ? 'left' : 'right'}
          visible={open}
          key={currentItem.id}
          goToChild={this.goToChild}
          goBack={this.goBack}
          hasParent={currentItem.id !== 'root'}
          toggleMenu={toggleMenu}
          currentItem={currentItem}
          backLabel={backLabel}
          animate={currentItem.id !== 'root' || !toChild}
          {...rest}
        />
      </div>
    );
  }
}

MobileMenu.propTypes = {
  content: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  toggleMenu: PropTypes.func.isRequired,
  defaultBackLabel: PropTypes.string.isRequired,
};

MobileMenu.defaultProps = {
  defaultBackLabel: '',
};

export default MobileMenu;
