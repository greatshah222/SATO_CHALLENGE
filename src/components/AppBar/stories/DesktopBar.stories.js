import React from 'react';
import { storiesOf } from '@storybook/react';
import { muiTheme } from 'storybook-addon-material-ui';
import { withKnobs, select, boolean } from '@storybook/addon-knobs';
import DesktopBar from '../DesktopBar.js';
import theme from '../../../theme';
import { accountItems, buttons, links, localeLink } from './menuItems';
import { bg } from './AppBar.stories';
import { action } from '@storybook/addon-actions';
import { suggestions } from '../../SiteSearch/stories/suggestions';

storiesOf('DesktopBar', module)
  .addDecorator(bg)
  .addDecorator(muiTheme(Object.values(theme)))
  .addDecorator(withKnobs)
  .addWithJSX('unsigned', () => (
    <DesktopBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={false}
      links={links}
      localeLink={localeLink}
      accountBtnLabel={'Kirjaudu'}
      accountBtnHandler={action('Login clicked')}
      searchSuggestions={suggestions}
      handleSearch={action('Search term:')}
      searchPlaceholder={'Etsi tietoa SATOlta...'}
      searchTerm={''}
      logoBtnHandler={action('Logo clicked')}
    />
  ))
  .addWithJSX('logged in', () => (
    <DesktopBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={true}
      links={links}
      localeLink={localeLink}
      accountBtnLabel={'Omasato'}
      accountBtnHandler={action('Omasato clicked')}
      searchSuggestions={[]}
      handleSearch={action('Search term:')}
      searchPlaceholder={'Etsi tietoa SATOlta...'}
      searchTerm={''}
      logoBtnHandler={action('Logo clicked')}
    />
  ))
  .addWithJSX('empty search results', () => (
    <DesktopBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={false}
      links={links}
      localeLink={localeLink}
      accountBtnLabel={'Kirjaudu'}
      accountBtnHandler={action('Login clicked')}
      searchSuggestions={[]}
      handleSearch={action('Search term:')}
      searchPlaceholder={'Etsi tietoa SATOlta...'}
      searchTerm={''}
      logoBtnHandler={action('Logo clicked')}
    />
  ))
  .addWithJSX('with search results', () => (
    <DesktopBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={false}
      links={links}
      localeLink={localeLink}
      accountBtnLabel={'Kirjaudu'}
      accountBtnHandler={action('Login clicked')}
      searchSuggestions={suggestions}
      handleSearch={action('Search term:')}
      searchPlaceholder={'Etsi tietoa SATOlta...'}
      searchTerm={''}
      logoBtnHandler={action('Logo clicked')}
    />
  ))
  .addWithJSX('playground', () => (
    <DesktopBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={boolean('Logged in', false, 'General')}
      links={links}
      localeLink={localeLink}
      accountBtnLabel={'Kirjaudu'}
      accountBtnHandler={action('Login clicked')}
      searchSuggestions={[]}
      handleSearch={action('Search term:')}
      searchPlaceholder={'Etsi tietoa SATOlta...'}
      searchTerm={''}
      logoBtnHandler={action('Logo clicked')}
    />
  ));
