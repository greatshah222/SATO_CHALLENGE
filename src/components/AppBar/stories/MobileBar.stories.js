import React from 'react';
import { storiesOf } from '@storybook/react';
import { muiTheme } from 'storybook-addon-material-ui';
import { withKnobs, select, boolean } from '@storybook/addon-knobs';
import { withViewport } from '@storybook/addon-viewport';
import { accountItems, buttons, links, localeLink } from './menuItems';
import { action } from '@storybook/addon-actions';
import { suggestions } from '../../SiteSearch/stories/suggestions';

import MobileBar from '../MobileBar.js';
import theme from '../../../theme';

import { bg } from './AppBar.stories';

storiesOf('MobileBar', module)
  .addDecorator(bg)
  .addDecorator(muiTheme(Object.values(theme)))
  .addDecorator(withKnobs)
  .addDecorator(withViewport('iphone6'))
  .addWithJSX('unsigned', () => (
    <div>
      <MobileBar
        accountItems={accountItems}
        buttons={buttons}
        loggedIn={false}
        links={links}
        localeLink={localeLink}
        defaultBackLabel={'Takaisin'}
        accountBtnLabel={'Kirjaudu'}
        accountBtnHandler={action('Login clicked')}
        searchSuggestions={[]}
        handleSearch={action('Search term:')}
        searchPlaceholder={'Etsi tietoa SATOlta...'}
        searchTerm={''}
        logoBtnHandler={action('Logo clicked')}
      />
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
      CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT CONTENT
    </div>
  ))
  .addWithJSX('with search results', () => (
    <MobileBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={false}
      links={links}
      localeLink={localeLink}
      defaultBackLabel={'Takaisin'}
      accountBtnLabel={'Kirjaudu'}
      accountBtnHandler={action('Login clicked')}
      searchSuggestions={suggestions}
      handleSearch={action('Search term:')}
      searchPlaceholder={'Etsi tietoa SATOlta...'}
      searchTerm={'vuokra'}
      logoBtnHandler={action('Logo clicked')}
    />
  ))
  .addWithJSX('logged in', () => (
    <MobileBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={true}
      links={links}
      localeLink={localeLink}
      defaultBackLabel={'Takaisin'}
      accountBtnLabel={'Omasato'}
      accountBtnHandler={action('Omasato clicked')}
      searchSuggestions={[]}
      handleSearch={action('Search term:')}
      searchPlaceholder={'Etsi tietoa SATOlta...'}
      searchTerm={''}
      logoBtnHandler={action('Logo clicked')}
    />
  ))
  .addWithJSX('playground', () => (
    <MobileBar
      accountItems={accountItems}
      buttons={buttons}
      loggedIn={boolean('Logged in', false, 'General')}
      links={links}
      localeLink={localeLink}
      defaultBackLabel={'Takaisin'}
      accountBtnLabel={'Kirjaudu'}
      accountBtnHandler={action('Login clicked')}
      searchSuggestions={[]}
      handleSearch={action('Search term:')}
      searchPlaceholder={'Etsi tietoa SATOlta...'}
      searchTerm={''}
      logoBtnHandler={action('Logo clicked')}
    />
  ));
