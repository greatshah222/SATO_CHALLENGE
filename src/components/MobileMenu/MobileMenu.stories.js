import React from 'react';
import { storiesOf } from '@storybook/react';
import { muiTheme } from 'storybook-addon-material-ui';
import { withKnobs, select, boolean } from '@storybook/addon-knobs';
import { withViewport } from '@storybook/addon-viewport';
import { buttons, links, localeLink } from '../AppBar/stories/menuItems';
import { action } from '@storybook/addon-actions';

import MobileMenu from './MobileMenu.js';
import theme from '../../theme';

storiesOf('MobileMenu', module)
  .addDecorator(muiTheme(Object.values(theme)))
  .addDecorator(withKnobs)
  .addDecorator(withViewport('iphone6'))
  .addWithJSX('default', () => (
    <MobileMenu
      content={{ items: [...buttons, ...links, localeLink], id: 'root' }}
      open={true}
      toggleMenu={action('toggleMenu')}
    />
  ))
  .addWithJSX('playground', () => (
    <MobileMenu
      content={{ items: [...buttons, ...links, localeLink], id: 'root' }}
      open={true}
      toggleMenu={action('toggleMenu')}
    />
  ));
