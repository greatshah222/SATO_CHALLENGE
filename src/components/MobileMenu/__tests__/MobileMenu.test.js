import React from 'react';
import { shallow } from 'enzyme';
import MobileMenu from '../MobileMenu';
import { buttons, links, localeLink } from '../../AppBar/stories/menuItems';

describe('MobileMenu', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <MobileMenu
        content={{ items: [...buttons, ...links, localeLink], id: 'root' }}
        open={true}
        toggleMenu={jest.fn()}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
