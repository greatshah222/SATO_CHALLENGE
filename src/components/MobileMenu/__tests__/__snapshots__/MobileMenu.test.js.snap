// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu renders correctly 1`] = `
<div>
  <WithStyles(MenuPane)
    animate={false}
    backLabel=""
    content={
      Object {
        "id": "root",
        "items": Array [
          Object {
            "active": true,
            "handler": [Function],
            "id": 1,
            "label": "Löydä koti",
            "variant": "button",
          },
          Object {
            "active": true,
            "handler": [Function],
            "id": 2,
            "items": Array [
              Object {
                "handler": [Function],
                "id": 6,
                "items": Array [
                  Object {
                    "handler": [Function],
                    "id": 9,
                    "items": Array [],
                    "label": "Kodin etsiminen",
                    "variant": "buttonMenuItem",
                  },
                ],
                "label": "Usein kysyttyä",
                "variant": "buttonMenuItem",
              },
              Object {
                "active": true,
                "handler": [Function],
                "id": 7,
                "label": "Ajankohtaista",
                "variant": "buttonMenuItem",
              },
              Object {
                "handler": [Function],
                "id": 8,
                "label": "Kotona",
                "variant": "buttonMenuItem",
              },
            ],
            "label": "Tietoa asumisesta",
            "variant": "button",
          },
          Object {
            "handler": [Function],
            "id": 4,
            "label": "Ota yhteyttä",
            "variant": "button",
          },
          Object {
            "active": true,
            "label": "Etusivu",
            "route": "/",
          },
          Object {
            "handler": [Function],
            "label": "Sijoittajille",
            "type": "smallOnMobile",
          },
          Object {
            "executeHandler": true,
            "handler": [Function],
            "id": 10,
            "items": Array [
              Object {
                "handler": [Function],
                "label": "Mediapankki",
                "variant": "buttonMenuItem",
              },
              Object {
                "handler": [Function],
                "label": "Etsitkö asiantuntijaa?",
                "variant": "buttonMenuItem",
              },
            ],
            "label": "Medialle",
            "small": true,
            "type": "smallOnMobile",
          },
          Object {
            "handler": [Function],
            "label": "SATO yrityksenä",
            "type": "smallOnMobile",
          },
          Object {
            "handler": [Function],
            "label": "In English",
          },
        ],
      }
    }
    currentItem={
      Object {
        "id": "root",
        "items": Array [
          Object {
            "active": true,
            "handler": [Function],
            "id": 1,
            "label": "Löydä koti",
            "variant": "button",
          },
          Object {
            "active": true,
            "handler": [Function],
            "id": 2,
            "items": Array [
              Object {
                "handler": [Function],
                "id": 6,
                "items": Array [
                  Object {
                    "handler": [Function],
                    "id": 9,
                    "items": Array [],
                    "label": "Kodin etsiminen",
                    "variant": "buttonMenuItem",
                  },
                ],
                "label": "Usein kysyttyä",
                "variant": "buttonMenuItem",
              },
              Object {
                "active": true,
                "handler": [Function],
                "id": 7,
                "label": "Ajankohtaista",
                "variant": "buttonMenuItem",
              },
              Object {
                "handler": [Function],
                "id": 8,
                "label": "Kotona",
                "variant": "buttonMenuItem",
              },
            ],
            "label": "Tietoa asumisesta",
            "variant": "button",
          },
          Object {
            "handler": [Function],
            "id": 4,
            "label": "Ota yhteyttä",
            "variant": "button",
          },
          Object {
            "active": true,
            "label": "Etusivu",
            "route": "/",
          },
          Object {
            "handler": [Function],
            "label": "Sijoittajille",
            "type": "smallOnMobile",
          },
          Object {
            "executeHandler": true,
            "handler": [Function],
            "id": 10,
            "items": Array [
              Object {
                "handler": [Function],
                "label": "Mediapankki",
                "variant": "buttonMenuItem",
              },
              Object {
                "handler": [Function],
                "label": "Etsitkö asiantuntijaa?",
                "variant": "buttonMenuItem",
              },
            ],
            "label": "Medialle",
            "small": true,
            "type": "smallOnMobile",
          },
          Object {
            "handler": [Function],
            "label": "SATO yrityksenä",
            "type": "smallOnMobile",
          },
          Object {
            "handler": [Function],
            "label": "In English",
          },
        ],
      }
    }
    defaultBackLabel=""
    direction="left"
    goBack={[Function]}
    goToChild={[Function]}
    hasParent={false}
    key="root"
    toggleMenu={[MockFunction]}
    visible={true}
  />
</div>
`;
